dms/models/dms_file.py

```diff
@@ -133,6 +133,27 @@ class File(models.Model):
         attachment=True, string="Content File", prefetch=False, invisible=True
     )
 
+    @api.depends("res_mimetype")
+    def _compute_custom_thumbnail_editable(self):
```

Comment:
This seems a bit convoluted/duplicated. We have the custom thumbnail field for adding any kind of preview according mimetypes. Why adding other mechanism?

Reply:
In line with https://github.com/OCA/dms/pull/121#discussion_r673703104, the purpose is to avoid using those computed thumbnails at all.

The fact of using a icon URL instead of an icon thumbnail improves performance a lot everywhere. For instance, your browser will only download one "folder.svg" icon when you're browsing your folders, instead of one icon per record. Besides, it'll cache it, so actually using different sizes is more slow & expensive (more resources to cache) than always using the big image.

OTOH I considered using the custom thumbnail fields as computed writable, `custom_thumbnail` is stored with `attachment=False`, so I couldn't just rely on Odoo's built-in filestore deduplication system if I applied the same image to the `custom_thumbnail` field. So, if I stored in the `custom_thumbnail` the same image as the file content, I'd be occupying 2x disk space: once on db and once on filestore. Well, actually more than 2x because of the 2 smaller thumbnail versions, but that's how thumbnails work so it's no the biggest problem.

Maybe a better fix would be to add `attachment=True` and write the corresponding migration script.

I preferred to keep it as it was because I didn't understand why it was `attachment=False` (so I assumed it was for a reason). Also, those custom smaller thumbnails would produce smaller versions, but then there's the caching and optimizations explained above...

So... well, things aren't so simple. I think we could rethink this on migration for v15 and remove garbage stuff in this module.

I'd *really* love to just get rid of all the custom thumbnails stuff. People can put tags and categories to files, so as not to need having those custom thumbnails. Besides, I doubt anybody in the world will ever takes screenshots of their PDFs to put them as thumbnails here. The use case is too hypothetical IMHO. Maybe we can remove them in v15 and, if there's a real need, let somebody else rescue code from v14 in a separate module?

Reply:
The requirement of custom preview is required for the purposes we have already talked about (other extensions providing such preview). The way is implemented is ours. There's no reason for `attachment=False` IMO, just to not remember such feature. Just think on how to apply KISS and DRY principles here.

Reply:
OK, do you have any example around about how to migrate from `attachment=False` to `attachment=True`? That'd help a lot.

Reply:
https://github.com/OCA/bank-payment/blob/10.0/account_banking_mandate/migrations/10.0.2.0.0/post-migration.py